src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityBuffer.java

```diff
@@ -120,20 +157,123 @@ public ModularPanel buildUI(PosGuiData guiData, PanelSyncManager guiSyncManager)
-                        .matrix(tankWidgets));
+                        .matrix(tankWidgets))
+                .child(Flow.row().pos(7, 18 * invTier + 23).width(36).height(18)
+                        .child(new ToggleButton()
+                                .value(new BoolValue.Dynamic(workingStateValueItems::getBoolValue,
+                                        workingStateValueItems::setBoolValue))
+                                .overlay(GTGuiTextures.BUTTON_ITEM_OUTPUT))
+                        .child(new ToggleButton()
+                                .value(new BoolValue.Dynamic(workingStateValueFluids::getBoolValue,
+                                        workingStateValueFluids::setBoolValue))
+                                .overlay(GTGuiTextures.BUTTON_FLUID_OUTPUT)));
     }
 
     @Override
     public int getTier() {
         return tier;
     }
 
+    public boolean isAutoOutputItems() {
+        return autoOutputItems;
+    }
+
+    public boolean isAutoOutputFluids() {
+        return autoOutputFluids;
+    }
+
+    public void setAutoOutputItems(boolean autoOutputItems) {
+        this.autoOutputItems = autoOutputItems;
+        if (!getWorld().isRemote) {
+            writeCustomData(UPDATE_AUTO_OUTPUT_ITEMS, buf -> buf.writeBoolean(autoOutputItems));
+            markDirty();
+        }
+    }
+
+    public void setAutoOutputFluids(boolean autoOutputFluids) {
+        this.autoOutputFluids = autoOutputFluids;
+        if (!getWorld().isRemote) {
+            writeCustomData(UPDATE_AUTO_OUTPUT_FLUIDS, buf -> buf.writeBoolean(autoOutputFluids));
+            markDirty();
+        }
+    }
+
+    @Override
+    public boolean onWrenchClick(EntityPlayer playerIn, EnumHand hand, EnumFacing facing,
+                                 CuboidRayTraceResult hitResult) {
+        if (!playerIn.isSneaking()) {
+            if (getOutputFacingItems() == facing) return false;
+            if (hasFrontFacing() && facing == getFrontFacing()) return false;
+            if (!getWorld().isRemote) {
+                setOutputFacing(facing);
+            }
+            return true;
+        }
+        return super.onWrenchClick(playerIn, hand, facing, hitResult);
+    }
+
+    public void setOutputFacing(EnumFacing outputFacing) {
+        this.outputFacingItems = outputFacing;
+        this.outputFacingFluids = outputFacing;
+        if (!getWorld().isRemote) {
+            notifyBlockUpdate();
+            writeCustomData(UPDATE_OUTPUT_FACING, buf -> {
+                buf.writeByte(outputFacingItems.getIndex());
+                buf.writeByte(outputFacingFluids.getIndex());
+            });
+            markDirty();
+        }
+    }
+
+    @Override
+    public void setFrontFacing(EnumFacing frontFacing) {
+        super.setFrontFacing(frontFacing);
+        if (this.outputFacingItems == null || this.outputFacingFluids == null) {
+            setOutputFacing(frontFacing.getOpposite());
+        }
```

Comment:
Why change the output facing when the front facing is changed? I do not believe any other of our machines do this.